### PR TITLE
Pin back to master and bump prometheus-operator to 0.41.1

### PR DESCRIPTION
**Pull request by lilic**
cc @coreos/team-monitoring

### CODE REVIEW

**Comment by s-urbaniak** [comment]
funky, there is no prometheus being deployed :thinking:

**Comment by s-urbaniak** [comment]
nevermind, i wasn't looking close enough, investigating too :thinking:

**Comment by s-urbaniak** [comment]
ah, here we go
```
monitoring           prometheus-k8s-0                             2/3     InvalidImageName   0          47s
```

**Comment by s-urbaniak** [comment]
```
  Warning  InspectFailed  35s (x3 over 70s)  kubelet, kind-control-plane  Failed to apply default image tag "quay.io/prometheus/prometheus:v2.20.0:v2.20.0": couldn't parse image reference "quay.io/prometheus/prometheus:v2.20.0:v2.20.0": invalid reference format
```

**Comment by s-urbaniak** [comment]
yeah the `prometheus` yaml seems to duplicate the version:
```
manifests/prometheus-prometheus.yaml
14:  image: quay.io/prometheus/prometheus:v2.20.0
34:  version: v2.20.0
```

**Comment by lilic** [comment]
Seems like the problem was [this PR](https://github.com/coreos/prometheus-operator/pull/3358/files) removed the way the full image is created, it is now generated from image name and tag, I fixed it in this in commit -> https://github.com/coreos/kube-prometheus/pull/628/commits/16b769c2fc3a1258c487d84cf8a44e579632602e

@brancz Before merging this, I am not sure is this want we wanted and expected from the above PR? Pinging you as you reviewed it and had some concerns on it.

**Comment by brancz** [comment]
hmm no this does look like a bug in prometheus operator .. it looks like the edge case that we are hitting [here](https://github.com/coreos/kube-prometheus/pull/628/files#diff-9c5516c9bc0931ed6067ac3d7cd2addfR34) is setting all three things like this:

```
version: v2.20.0
tag: v2.20.0
image: quay.io/prometheus/prometheus:v2.20.0
```

**Comment by lilic** [comment]
Yes exactly my thinking, I think if image has tag it should not be set again, or do we blame that on user error?

**Comment by brancz** [comment]
note, the reasoning behind all of these changes is to make `version` less overloaded in responsibilities (in the long run) .. it was a mistake in the beginning to make it both the tag of the image as well as the "branching-factor" to decide how to provision prometheus according to the version. What I would like to see long term is making `version` only the latter, and have `image` be the only thing that decides the image.

**Comment by lilic** [comment]
Agreed, I think version and tag/image should not be taken into consideration.

File: jsonnet/kube-prometheus/prometheus/prometheus.libsonnet [hunk 1]
```diff
@@ -178,7 +178,8 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
         spec: {
           replicas: p.replicas,
           version: $._config.versions.prometheus,
-          image: $._config.imageRepos.prometheus + ':' + $._config.versions.prometheus,
+          image: $._config.imageRepos.prometheus,
+	  tag: $._config.versions.prometheus,
```
Comment (brancz) [inline jsonnet/kube-prometheus/prometheus/prometheus.libsonnet:182]: we should just remove this I think, then it should all work with specifying the tag in the `image`, as ultimately we want to deprecate `tag`, `sha` and `baseImage`

Comment (lilic) [reply jsonnet/kube-prometheus/prometheus/prometheus.libsonnet]: Done, can you have another look, thanks!

File: jsonnet/kube-prometheus/prometheus/prometheus.libsonnet [hunk 2]
```diff
@@ -178,7 +178,7 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
         spec: {
           replicas: p.replicas,
           version: $._config.versions.prometheus,
-          image: $._config.imageRepos.prometheus + ':' + $._config.versions.prometheus,
+          image: $._config.imageRepos.prometheus,
```
Comment (brancz) [inline jsonnet/kube-prometheus/prometheus/prometheus.libsonnet:181]: `image` should be the full image as it was before, if that doesn't work, then we need to fix the prometheus operator in that regard, I think it was the combination of specifying `version`, `tag`, and the full `image` that caused the edge case to trigger before

Comment (lilic) [reply jsonnet/kube-prometheus/prometheus/prometheus.libsonnet:181]: Full image with tag in the image does not work, as the version gets added to it, hence the initial failure, after removing that and adding either tag or leaving version alone that fixes it. We did not have tag before, just image and version.

Comment (brancz) [reply jsonnet/kube-prometheus/prometheus/prometheus.libsonnet:181]: Then this is broken and needs to be fixed in prometheus operator

Comment (lilic) [reply jsonnet/kube-prometheus/prometheus/prometheus.libsonnet]: No, sorry for not being clear enough it was just full `image` and `version`. I reverted the changes to rerun the CI again.

Comment (lilic) [reply jsonnet/kube-prometheus/prometheus/prometheus.libsonnet]: Yep fails, so we fix it in master and latest release before we can bump this?

Comment (brancz) [reply jsonnet/kube-prometheus/prometheus/prometheus.libsonnet]: yes

**Comment by lilic** [comment]
Blocked by https://github.com/prometheus-operator/prometheus-operator/pull/3392

**Comment by lilic** [comment]
No worries at all! Regression was fixed in prom-op 🎉

**Pull request merged**